### PR TITLE
Attach target instance to source composable for all defined target types

### PR DESCRIPTION
**Pull request by mkeckmkeck**
This makes autotransform auto-attaching 

Fixes: SE-9805

### CODE REVIEW

File: src/main/java/sirius/kernel/di/transformers/AutoTransformLoadAction.java [hunk 1]
```diff
@@ -147,13 +160,25 @@ public void handle(@Nonnull MutableGlobalContext ctx, @Nonnull Class<?> clazz) t
...
+            ctx.registerPart(new AutoTransformer<>(clazz, autoTransform, target, allTargets), Transformer.class);
         }
+    }
+
+    private Class<?>[] mergeAllTargets(AutoTransform autoTransform) {
         Class<?> target = autoTransform.target();
         if (!target.equals(Object.class)) {
-            ctx.registerPart(new AutoTransformer<>(clazz, autoTransform, target), Transformer.class);
+            Class<?>[] targets = autoTransform.targets();
+            Class<?>[] mergedTargets = Arrays.copyOf(targets, targets.length + 1);
```
Comment (idlira) [inline src/main/java/sirius/kernel/di/transformers/AutoTransformLoadAction.java:178]: Would this add double entries in the target list?

Comment (mkeckmkeck) [reply src/main/java/sirius/kernel/di/transformers/AutoTransformLoadAction.java:178]: Yes, the array could contain duplicates if the AutoTransform annotation defines them. Those will later be catched at sirius.kernel.di.transformers.Composable#attach(java.lang.Class<? extends T>, T) where the last caller wins. With sane usage, there should be no problem.

File: src/main/java/sirius/kernel/di/transformers/AutoTransformLoadAction.java [hunk 2]
```diff
@@ -147,13 +160,25 @@ public void handle(@Nonnull MutableGlobalContext ctx, @Nonnull Class<?> clazz) t
...
+        Class<?>[] allTargets = mergeAllTargets(autoTransform);
         Class<?>[] targets = autoTransform.targets();
         for (Class<?> target : targets) {
-            ctx.registerPart(new AutoTransformer<>(clazz, autoTransform, target), Transformer.class);
+            ctx.registerPart(new AutoTransformer<>(clazz, autoTransform, target, allTargets), Transformer.class);
+        }
+        Class<?> target = autoTransform.target();
+        if (!target.equals(Object.class)) {
+            ctx.registerPart(new AutoTransformer<>(clazz, autoTransform, target, allTargets), Transformer.class);
         }
```
Comment (qw3ry) [inline src/main/java/sirius/kernel/di/transformers/AutoTransformLoadAction.java:171]: Warum nicht direkt über die mergedTargets drüber iterieren?

Comment (mkeckmkeck) [reply src/main/java/sirius/kernel/di/transformers/AutoTransformLoadAction.java:171]: Guter Punkt, fix ich gleich

Comment (mkeckmkeck) [reply src/main/java/sirius/kernel/di/transformers/AutoTransformLoadAction.java]: done

**Pull request merged**